src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/MSBuildRestoreUtility.cs

```diff
@@ -228,12 +229,38 @@ public static PackageSpec GetPackageSpec(IEnumerable<IMSBuildItem> items)
                 {
                     // Add RIDs and Supports
                     result.RuntimeGraph = GetRuntimeGraph(specItem);
+
+                    // Add PackageTargetFallback
+                    AddPackageTargetFallbacks(result, items);
                 }
             }
 
             return result;
         }
 
+        private static void AddPackageTargetFallbacks(PackageSpec spec, IEnumerable<IMSBuildItem> items)
+        {
+            foreach (var item in GetItemByType(items, "TargetFrameworkInformation"))
+            {
+                var frameworkString = item.GetProperty("TargetFramework");
+                TargetFrameworkInformation targetFrameworkInfo = null;
+
+                if (!string.IsNullOrEmpty(frameworkString))
+                {
+                    targetFrameworkInfo = spec.GetTargetFramework(NuGetFramework.Parse(frameworkString));
+                }
+
+                Debug.Assert(targetFrameworkInfo != null, $"PackageSpec does not contain the target framework: {frameworkString}");
+
+                if (targetFrameworkInfo != null)
+                {
+                    targetFrameworkInfo.Imports = Split(item.GetProperty("PackageTargetFallback"))
+                        .Select(NuGetFramework.Parse)
```

Comment:
It would get the same behavior as before once the package spec is read, and I believe we filter them out. The long term plan is for this to go away and be replaced by TFM mappings.